src/subscription_manager/utils.py

```diff
@@ -482,7 +499,7 @@ def match(self, cert):
 
 
 class EntitlementCertificateFilter(ProductCertificateFilter):
-    def __init__(self, filter_string=None, service_level=None):
+    def __init__(self, filter_string: str = None, service_level=None):
```

Comment:
i guess `Optional[str]` for `filter_string`? also, did you forget `service_level` here? :)